src/epoch.rs

```diff
@@ -2226,6 +2226,82 @@ impl Epoch {
         me
     }
 
+    /// Builds an Epoch from given `wk` week counter into the desired Time scale.
+    /// `ns` is the amount of nanoseconds into that week from closest Sunday Midnight.
+    /// This is usually how GNSS receivers describe a GNSS time scale epoch
+    #[cfg(feature = "python")]
+    #[staticmethod]
+    pub fn from_timeofweek(wk: u32, ns: u64, ts: TimeScale) -> Self {
+        let week = Duration::from_seconds(wk as f64 * SECONDS_PER_DAY * 7.0);
+        Self::from_duration(week + (ns as f64) * Unit::Nanosecond, ts)
+    }
+
+    #[cfg(feature = "python")]
+    #[staticmethod]
+    pub fn from_timeofweek_utc(wk: u32, ns: u64) -> Self {
+        let week = Duration::from_seconds(wk as f64 * SECONDS_PER_DAY * 7.0);
+        Self::from_utc_duration(week + (ns as f64) * Unit::Nanosecond)
+    }
+
+    /// Converts to "time of week" (`wk, `tow`),
+    /// which is usually how GNSS receivers describe a timestamp.
+    /// `wk` is a rolling week conter into that time scale,
+    /// `tow` is the number of seconds since closest Sunday midnight into that week.
+    pub fn to_timeofweek_utc(&self) -> (u32, u64) {
+        // fractionnal days in this time scale
+        let days = match self.time_scale {
+            TimeScale::GPST => self.to_gpst_days(),
+            TimeScale::GST => self.to_gst_days(),
+            TimeScale::BDT => self.to_bdt_days(),
+            TimeScale::TT => self.to_tt_days(),
+            TimeScale::TAI => self.to_tai_days(),
+            TimeScale::UTC => self.to_utc_days(),
+            TimeScale::TDB => self.to_tdb_seconds() / SECONDS_PER_DAY,
+            TimeScale::ET => self.to_et_seconds() / SECONDS_PER_DAY,
+        };
+        // wk: rolling week counter into timescale
+        let wk = (days / 7.0) as u32;
+        // tow: number of nanoseconds between self and closest sunday midnight / start of week
+        let start_of_week = self.closest_utc_start_of_week();
+        let dw = *self - start_of_week; // difference in weekdays [0..6]
+        let (_, d, h, m, s, ms, us, ns) = dw.decompose();
+        let tow = Duration::from_days(d as f64)
```

Comment:
You can use `Duration::compose(...)` here with the same parameters that `decompose` returns. Although I don't quite understand why you call `decompose` if you rebuild the Duration with all of the elements from `decompose` just afterwards. It seems like you could just do:

```suggestion
        let (_, mut tow) = dw.to_parts();
```

But maybe I'm misunderstanding something.